lib/src/modules/apply.cc

```diff
@@ -225,43 +358,55 @@ exit exit_code
 
     Util::CommandObject Apply::buildCommandObject(const ActionRequest& request)
     {
+        // Shared
         auto params = request.params();
+        params.set<std::string>("ca", ca_);
+        params.set<std::string>("crt", crt_);
+        params.set<std::string>("key", key_);
+        params.set<std::string>("crl", crl_);
+        params.set<std::string>("proxy", proxy_);
+
         const fs::path& results_dir { request.resultsDir() };
-        // Ensure the ruby shim is in the cache dir.
-        // TODO: remove this when we figure out how to manage ruby shim
-        const std::string ruby_shim_cache_dir = "apply_ruby_shim";
-        const std::string ruby_shim_script_name = "apply_ruby_shim.rb";
+
+        const std::string ruby_shim_cache_dir = "apply_ruby_shims";
         const fs::path& cache_dir = module_cache_dir_->createCacheDir(ruby_shim_cache_dir);
-        auto ruby_shim_path = cache_dir / ruby_shim_script_name;
-        if (!fs::exists(ruby_shim_path)) {
-          lth_file::atomic_write_to_file(RUBY_SHIM, ruby_shim_path.string(), NIX_DOWNLOADED_FILE_PERMS, std::ios::binary);
+
```

Comment:
Does this fit with general module design? There are no other built-in cpp modules that have multiple actions implemented for a single module. Do we just handle the logic in buildCommandObject? Or are we supposed to have separate implementations of buildCommandObject for each action?

Reply:
I think this is the first module with multiple actions, so we're in new territory here.

Personally I think we should stick with one buildCommandObject until someone sees a good reason to split it